src/game.cpp

```diff
@@ -2180,16 +2188,27 @@ bool Game::connectToServer(const std::string &playername,
 				break;
 			}
 
-			wait_time += dtime;
-			// Only time out if we aren't waiting for the server we started
-			if (!address->empty() && wait_time > 10) {
-				*error_message = "Connection timed out.";
-				errorstream << *error_message << std::endl;
-				break;
-			}
+			if (client->m_is_registration_confirmation_state) {
+				if (registration_confirmation_shown) {
+					// Keep drawing the GUI
+					RenderingEngine::draw_menu_scene(guienv, dtime, true);
+				} else {
+					registration_confirmation_shown = true;
+					(new GUIConfirmRegistration(guienv, guienv->getRootGUIElement(), -1,
+						   &g_menumgr, client, playername, password, *address, aborted))->drop();
+				}
```

Comment:
what happens if the user waits until the connection is timed out (a minute or such) and then clicks OK?

Reply:
The connection will not be timed out when showing this dialog. After player click on "Register and Join", it will start authenticate to server.

https://github.com/srifqi/minetest/blob/e78e694a421c0deb84dff43c9d984092826a970b/src/game.cpp#L2199-L2210

Reply:
How about a server-sided connection reset for the connecting player? (crash, shutdown, network issues, etc)

Reply:
Haven't tried it.

Reply:
Tried. It will shown as "Connection timed out."